doc/06-anonymous-tracking.md

```diff
@@ -0,0 +1,65 @@
+
+On web, the affected user and session identifiers are stored in the cookies.
+
+Concretely, the following user and session identifiers can be anonymised:
+
+* Mobile client-side user identifiers: the `userId` in Session context entity and the IDFA identifiers (`appleIdfa` and `appleIdfv`, and their Android equivalents) in the Platform context entity.
+* Mobile client-side session identifiers: `sessionId` in Session context entity.
+* Web client-side user identifiers: the `domain_userid` in the sp_id cookie.
+* Web client-side session identifiers: `sessionId` in Session context entity, `domain_sessionid`, and `domain_sessionidx` from the sp_id cookie.
+* Server-side user identifiers: `network_userid` and `user_ipaddress` event properties.
+
+There are several levels to the anonymisation depending on which of the three categories are affected:
+
+## 1. Full client-side anonymisation
+
+In this case, we want to anonymise both the client-side user identifiers as well as the client-side session identifiers.
+
+On mobile, setting `userAnonymisation` affects properties in the Session and Platform entities. If neither is present, it has no effect. In the Platform context entity, the IDFA identifiers are anonymised. Here the Session entity is not configured: no session information will be tracked at all.
+
+```dart
+const TrackerConfiguration()
+    .userAnonymisation(true)
+    .sessionContext(false)
+    .platformContext(true) // only relevant for mobile
+```
+
+On web, setting `userAnonymisation` stops any user identifiers or session information being stored in cookies or localStorage. This means that `domain_userid`, `domain_sessionid`, and `domain_sessionidx` will be anonymised. Again, no session information will be tracked at all.
+
+```dart
+const TrackerConfiguration().userAnonymisation(true)
+```
+
+## 2. Client-side anonymisation with session tracking
+
+For mobile only, this setting disables client-side user identifiers but tracks session information.
+
+```dart
+const TrackerConfiguration()
+    .userAnonymisation(true)
+    .sessionContext(true)
+    .platformContext(true)
+```
+On mobile, enabling both `userAnonymisation` and `sessionContext` means that events will have the Session context entity, but the `userId` property will be anonymised to a null UUID (`00000000-0000-0000-0000-000000000000`). As above, if the Platform context entity is enabled, the IDFA identifiers will not be present.
+
+On web, enabling both `userAnonymisation` and `sessionContext` is the same as enabling just `userAnonymisation`; the Session entity is not added to events.
```

Comment:
I didn't realize this before, but I think you're right, the session context entity is not added to events if user anonymisation is enabled. I created [an issue on the JS tracker repo](https://github.com/snowplow/snowplow-javascript-tracker/issues/1124) to fix this as it is not the desired behaviour – it should work the same as on the mobile trackers.

I don't think we need to wait for the JS tracker issue to be solved, but we should build the Flutter tracker to expect that it will be resolved. Hopefully we can publish a new JS tracker release with the fix soon.

So I wouldn't note the behaviour here explicitly, or say that it is so until version 3.6 of the tracker and give a link to the JS tracker issue.